### PR TITLE
Refactoring to dynamic properties model structire

### PR DESCRIPTION
**Pull request by stasbar**
Took 4 hours 19 minutes

### CODE REVIEW

File: src/components/AffiliateEditTable/NewAffiliateModal.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { Form, Input, Modal } from 'antd';
 import * as React from 'react';
 import { FormComponentProps } from 'antd/es/form';
 import { connect } from 'dva';
-import { ConnectState, Dispatch } from '@/models/connect';
+import { ConnectState } from '@/models/connect';
+import { Dispatch } from 'redux';
```
Comment (Argeath) [inline src/components/AffiliateEditTable/NewAffiliateModal.tsx]: it shouldn't be from 'dva'?

File: src/components/BatteryView/index.tsx [hunk 2]
```diff
@@ -12,12 +13,7 @@ interface BatteryViewProps extends ConnectProps {
 }
 
 const BatteryView: React.FC<BatteryViewProps> = ({ battery, height, width, dispatch }) => {
-  const onCardClick = () =>
-    dispatch &&
-    dispatch({
-      type: 'batteries/selectBattery',
-      battery,
-    });
+  const onCardClick = () => dispatchSelectBattery(dispatch!, battery);
```
Comment (Argeath) [inline src/components/BatteryView/index.tsx]: remove "!"

File: src/components/ComplexWire/index.tsx [hunk 3]
```diff
@@ -80,27 +85,11 @@ const ComplexWire: React.FC<WireComponentProps> = props => {
...
+    key && dispatchSetCoilType(dispatch, WireType[key], path);
   const onPitchChange = (value: number | undefined) =>
-    value &&
-    dispatch &&
-    dispatch({
-      type: 'coil/setInnerDiameter',
-      payload: value,
-    });
+    value && dispatchSetInnerDiameter(dispatch, value);
   const onAddWireClick = () => {
```
Comment (Argeath) [inline src/components/ComplexWire/index.tsx]: Curly brackets not needed

File: src/components/GlobalHeader/AvatarDropdown.tsx [hunk 4]
```diff
@@ -6,80 +6,79 @@ import { connect } from 'dva';
...
 
-class AvatarDropdown extends React.Component<GlobalHeaderRightProps> {
-  onMenuClick = (event: ClickParam) => {
+const AvatarDropdown: React.FC<GlobalHeaderRightProps> = props => {
+  const onMenuClick = (event: ClickParam) => {
     const { key } = event;
 
     if (key === 'logout') {
-      this.props.dispatch!({ type: 'user/logout' });
+      dispatchLogout(props.dispatch!);
```
Comment (Argeath) [inline src/components/GlobalHeader/AvatarDropdown.tsx]: remove "!"

File: src/components/SingleWire/index.tsx [hunk 5]
```diff
@@ -52,21 +52,11 @@ const materials: Material[] = [
...
 
-  const handleMaterialChange = (): void => {
-  };
-  const onDeleteClick = () => {
-    dispatch({
-      type: 'coil/deleteWire',
-      paths: path,
-    });
-  };
+  const handleMaterialChange = (): void => {};
```
Comment (Argeath) [inline src/components/SingleWire/index.tsx]: Add TODO comment to indicate that's not finished yet.

File: src/components/UploadPost/index.tsx [hunk 6]
```diff
@@ -16,36 +23,18 @@ const UploadPost: React.FC<UploadPostProps> = props => {
...
-      title: e.target.value,
-    });
-  };
-  const capitalize = (s: string) => s.charAt(0).toUpperCase() + s.slice(1);
-  const onPostClick = () => {
-    dispatch({
-      type: `uploadPost/submit${capitalize(type)}`,
-    });
-  };
+  const onTextChange = (e: any) => dispatchSetText(dispatch, e.target.value);
```
Comment (Argeath) [inline src/components/UploadPost/index.tsx]: Don't use any.

File: src/models/operations.ts [hunk 7]
```diff
@@ -196,14 +199,13 @@ const OperationModel: OperationsModelType = {
     *deleteComment({ itemId, commentId, what }, { call }) {
       try {
         switch (what) {
-          case 'photo':
-          case 'gear':
```
Comment (Argeath) [inline src/models/operations.ts]: 'gear' not longed needed?

File: src/components/SingleWire/index.tsx (hunk 5 above)
Comment (stasbar) [reply src/components/SingleWire/index.tsx]: done

File: src/components/GlobalHeader/AvatarDropdown.tsx (hunk 4 above)
Comment (stasbar) [reply src/components/GlobalHeader/AvatarDropdown.tsx]: done

File: src/components/ComplexWire/index.tsx (hunk 3 above)
Comment (stasbar) [reply src/components/ComplexWire/index.tsx]: done

File: src/components/BatteryView/index.tsx (hunk 2 above)
Comment (stasbar) [reply src/components/BatteryView/index.tsx]: done

File: src/components/UploadPost/index.tsx (hunk 6 above)
Comment (stasbar) [reply src/components/UploadPost/index.tsx]: done

File: src/models/operations.ts (hunk 7 above)
Comment (stasbar) [reply src/models/operations.ts]: No it was mistake. Let's make the app level abstraction completely unaware of database namings. For the app, the Photos are just photos, app doesn't care if it's called 'gears' in database.

**Comment by stasbar** [comment]
@Argeath you can merge if everything is ok.

**Pull request merged**